### PR TITLE
Fix static drives 'a:, c:' not redirected

### PR DESCRIPTION
**Pull request by jlafontaine-devolutions**
Allow FreeRDP to correctly redirect static drives like 'A:' and 'C:'.

### CODE REVIEW

**Comment by bmiklautz** [comment]
@freerdp-bot test

**Comment by akallabeth** [comment]
@freerdp-bot test

**Comment by jlafontaine-devolutions** [comment]
I tested on all platforms and the expected drives were redirected. 
Windows now redirects my C drive and has the same redirected drives as mstsc.exe.

**Comment by akallabeth** [comment]
Will recheck and post the results tomorrow. Github does not like us today :/

**Comment by akallabeth** [comment]
@jlafontaine-devolutions I've checked with `+drives` (fedora 28):
1. Hotplug events are handled, ok
1. Initial population of mount points is not working, so only drives attached after the client is started are redirected/removed

**Comment by jlafontaine-devolutions** [comment]
@akallabeth 
I've double checked +drives on Ubuntu 18.04.1.
1. They are.
2. You are right, on linux, hotplugable drives don't show up upon connection. They do on Windows.
I will fix that.

**Comment by jlafontaine-devolutions** [comment]
Initial population is now working with the latest commit.

**Comment by akallabeth** [comment]
@freerdp-bot test

File: channels/rdpdr/client/rdpdr_main.c [hunk 1]
```diff
@@ -1089,8 +1061,6 @@ static UINT rdpdr_process_connect(rdpdrPlugin* rdpdr)
 					WLog_ERR(TAG, "CreateThread failed!");
 					return ERROR_INTERNAL_ERROR;
 				}
-
-				continue;
```
Comment (akallabeth) [inline channels/rdpdr/client/rdpdr_main.c]: Why remove this? Adding a pseudo device for a hotplug wildcard does not really make sense or does it?

File: channels/rdpdr/client/rdpdr_main.c [hunk 2]
```diff
@@ -1079,7 +1051,7 @@ static UINT rdpdr_process_connect(rdpdrPlugin* rdpdr)
 		{
 			RDPDR_DRIVE* drive = (RDPDR_DRIVE*)device;
 
-			if (drive->Path && (strcmp(drive->Path, "*") == 0))
+			if (!rdpdr->hotplugThread && drive->Path && (strcmp(drive->Path, "*") == 0))
```
Comment (akallabeth) [inline channels/rdpdr/client/rdpdr_main.c]: Might be good, if you allow more than one wildcard.
Still, adding wildcard devices (see below) does not seem to make sense.

File: channels/rdpdr/client/rdpdr_main.c (hunk 1 above)
Comment (jlafontaine-devolutions) [reply channels/rdpdr/client/rdpdr_main.c]: The first_hotplug and hotplug thread on Windows exclude fixed drives but adding the speudo device redirects all my drives correctly.
Would it be better if check_path allowed more drive types?

File: channels/rdpdr/client/rdpdr_main.c (hunk 2 above)
Comment (jlafontaine-devolutions) [reply channels/rdpdr/client/rdpdr_main.c]: That was some defensive programming. 
If only one wild card could ever happen, I can remove the check.

File: channels/rdpdr/client/rdpdr_main.c (hunk 1 above)
Comment (akallabeth) [reply channels/rdpdr/client/rdpdr_main.c]: yes, might be the most simple solution. If that works for you too.

**Comment by jlafontaine-devolutions** [comment]
Added drive_fixed in the list of valid hot plug devices.

**Comment by jlafontaine-devolutions** [comment]
Also pushed a similar fix for MacOS where Macintosh HD is a symlink to / and wasn't redirected.

**Pull request closed without merging**